### DISCUSSION TITLE
Unable to exclude nested fields with `jsonable_encoder`

### DISCUSSION THREAD

**Discussion by narmafraz** [fastapi/fastapi#8890] [Questions]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from typing import List

from fastapi.encoders import jsonable_encoder
from pydantic.main import BaseModel

class YourBaseModel(BaseModel):
    bar: int
    baz: int

class MyBaseModel(BaseModel):
    nest: List[YourBaseModel] = None
    foo: int

model = MyBaseModel(nest=[YourBaseModel(bar=2, baz=3), YourBaseModel(bar=4, baz=5)], foo=1)
exclude_pattern={'nest': {'__all__': {'baz'}}}
expected={"nest":[{"bar": 2}, {"bar": 4}], "foo": 1}

def test_pydantic_nested_exclude():
    assert model.dict(exclude=exclude_pattern) == expected # Passed

def test_jsonable_encoder_nested_exclude():
    assert jsonable_encoder(model, exclude=exclude_pattern) == expected # Failed
    # actual={'foo': 1} ## i.e. removes all nested elements!
```

### Description

* Define a nested model e.g. `{"nest":[{"bar": 2, baz=3}, {"bar": 4, baz=5}], "foo": 1}`
* Try to exclude one of the fields in the nested model (e.g. `baz`) using `jsonable_encoder`
* It excludes the whole nested field altogether i.e. `nest` and returns `{'foo': 1}`
* But I expected it to return `{"nest":[{"bar": 2}, {"bar": 4}], "foo": 1}`.

### Environment

* OS: Windows
* FastAPI Version: 0.65.1
* Python version: 3.8.2

### Additional context

I followed along with issue #2594 and it said the behaviour explained in https://pydantic-docs.helpmanual.io/usage/exporting_models/#advanced-include-and-exclude should work but this doesn't seem to be the case. 
What am I doing wrong? :<
Thanks in advance.

**Comment by frankie567** [comment]
Unfortunately, you seem to hit the bug #2007. There is a PR to fix it (#2016), but has not been merged yet.

**Comment by lovetoburnswhen** [comment]
Looks like it's been merged as of `v0.66.0`

**Comment by Spirit412** [comment]
DONT WORK: `response_model_include={..., ('is_api_user', 'order_client_c')}`
DUT WORK `response_model_include={..., *('is_api_user', 'order_client_c')}`

**Comment by cn4reg** [comment]
> ('is_api_user', 'order_client_c')

`rersponse_model_include` should be SetIntStr or DictIntStrAny or None, so I think 
this 

> response_model_include={..., ('is_api_user', 'order_client_c')}

could not work. cause `('is_api_user', 'order_client_c')` is an anonymous Set object which contains str.

And the latter one

> response_model_include={..., *('is_api_user', 'order_client_c')}

could works, this unpacks the str values in the Set object.

IMO, they are not the same thing.

**Comment by cn4reg** [comment]
And also, I would like to put a complete example here.

```python

import typing as t

from fastapi import APIRouter, FastAPI
from pydantic import BaseModel, Field


class Base(BaseModel):
    a: int = Field(default=0)
    b: str = Field(default="None")

    class Config:
        orm_mode = True


class ListBase(BaseModel):
    c: t.List[Base]


app = FastAPI(title="test", docs_url="/api/docs", openapi_url="/api")
r = APIRouter()


@r.get("/test",
       response_model=ListBase,
       response_model_exclude={"c": {"__all__": {"b"}}})
async def test_exclude():
    results = {"c": [{"a": 1, "b": "aa"}, {"a": 2, "b": "ba"}]}
    return results


app.include_router(r, prefix="/api")

```

Output:

```
{
  "c": [
    {
      "a": 1
    },
    {
      "a": 2
    }
  ]
}
````
